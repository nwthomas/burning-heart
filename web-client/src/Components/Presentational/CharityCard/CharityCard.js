import React from "react";

const CharityCard = ({ charity }) => {
  return (
    <div className="charity__card">
      <p className="charity__name">{charity.charityName}</p>
    </div>
  );
};

export default CharityCard;
