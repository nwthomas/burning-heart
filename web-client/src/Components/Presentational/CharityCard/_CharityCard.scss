.charity__card {
  width: 90%;
  border-bottom: 2px solid #cacacf;
  padding: 30px 0;
  margin: 0 5%;
  display: flex;
  justify-content: space-between;
  align-items: center;

  .charity__name {
    width: 50%;
  }

  .charity__donate-button {
    width: 25%;
    height: 40px;
    border-radius: 50px;
    border: 1px solid #cacacf;
    cursor: pointer;

    &:hover {
      background: #cacacf;
      color: #ffffff;
    }
  }
}

.charity__card:last-of-type {
  border-bottom: 0;
}
