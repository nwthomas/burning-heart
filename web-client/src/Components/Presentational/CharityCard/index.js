export { default as CharityCard } from "./CharityCard";
