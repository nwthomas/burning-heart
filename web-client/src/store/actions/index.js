import axios from "axios";
import decode from "jwt-decode";
import {
  MOVE_SIGNUP_FORM_FORWARD,
  MOVE_SIGNUP_FORM_BACKWARD,
  UPDATE_SIGNUP_FORM,
  UPDATE_DONOR_FORM,
  UPDATE_CHARITY_FORM,
  UPDATE_CHARITY_OWNER_FORM,
  CREATE_NEW_ACCOUNT_START,
  CREATE_NEW_ACCOUNT_SUCCESS,
  CREATE_NEW_ACCOUNT_ERROR,
  CREATE_NEW_CHARITY_START,
  CREATE_NEW_CHARITY_SUCCESS,
  CREATE_NEW_CHARITY_ERROR,
  CLOSE_SIGNUP_MODAL,
  UPDATE_LOGIN_FORM,
  LOGIN_APP_START,
  LOGIN_APP_SUCCESS,
  LOGIN_APP_ERROR,
  LOGIN_CHARITY_START,
  LOGIN_CHARITY_SUCCESS,
  LOGIN_CHARITY_ERROR,
  CLOSE_LOGIN_MODAL,
  LOGOUT_APP,
  EXPIRED_CREDENTIALS,
  LOGIN_TOKEN_START,
  LOGIN_TOKEN_SUCCESS,
  LOGIN_TOKEN_ERROR,
  LOGIN_TOKEN_CHARITY_START,
  LOGIN_TOKEN_CHARITY_SUCCESS,
  LOGIN_TOKEN_CHARITY_ERROR,
  FETCH_USER_DONATIONS_START,
  FETCH_USER_DONATIONS_SUCCESS,
  FETCH_USER_DONATIONS_ERROR,
  FETCH_CHARITY_DONATIONS_START,
  FETCH_CHARITY_DONATIONS_SUCCESS,
  FETCH_CHARITY_DONATIONS_ERROR,
  FETCH_CHARITIES_START,
  FETCH_CHARITIES_SUCCESS,
  FETCH_CHARITIES_ERROR,
  SELECT_DONATIONS,
  SELECT_CHARITIES,
  SELECT_CHARITY,
  CARD_PAYMENT_START,
  CARD_PAYMENT_SUCCESS,
  CARD_PAYMENT_ERROR,
  CLOSE_PAYMENT_MODAL,
  HANDLE_PAYMENT_FORM,
  HANDLE_CARD_TOKEN_ERROR
} from "../types";

const restrictedError = "Not authorized. Please try logging in again.";

const charityDetails = {
  type: "custom", // Preset
  country: "US", // Preset
  email: "",
  requested_capabilities: ["card_payments"], // Preset
  business_type: "company", // Preset
  company: {
    address: {
      city: "",
      line1: "",
      line2: "",
      postal_code: "",
      state: ""
    }
  },
  name: "", // Company name
  phone: "",
  tax_id: "",
  external_account: {
    object: "bank_account",
    country: "US",
    currency: "usd",
    routing_number: "",
    account_number: ""
  },
  individual: {
    // Finish
  }
};

//============================================================== Signup Action Creators
export const createDonorAccount = userDetails => dispatch => {
  dispatch({ type: CREATE_NEW_ACCOUNT_START });
  axios
    .post(
      "https://burning-heart.herokuapp.com/api/auth/register-account",
      userDetails
    )
    .then(res => {
      dispatch({ type: CREATE_NEW_ACCOUNT_SUCCESS, payload: res.data });
    })
    .catch(err => {
      dispatch({ type: CREATE_NEW_ACCOUNT_ERROR, payload: err });
    });
};

export const createCharityAccount = charityDetails => dispatch => {
  dispatch({ type: CREATE_NEW_CHARITY_START });
  axios
    .post("http://localhost:7000/api/auth/register-charity", charityDetails)
    .then(res => {
      dispatch({ type: CREATE_NEW_CHARITY_SUCCESS, payload: res.data });
    })
    .catch(err => {
      dispatch({ type: CREATE_NEW_CHARITY_ERROR, payload: err });
    });
};

export const handleSignupFormChanges = (name, value) => {
  return {
    type: UPDATE_SIGNUP_FORM,
    payload: { name, value }
  };
};

export const handleDonorFormChanges = (name, value) => {
  return {
    type: UPDATE_DONOR_FORM,
    payload: { name, value }
  };
};

export const handleCharityFormChanges = (name, value) => {
  return {
    type: UPDATE_CHARITY_FORM,
    payload: { name, value }
  };
};

export const handleCharityOwnerForm = (name, value) => {
  return {
    type: UPDATE_CHARITY_OWNER_FORM,
    payload: { name, value }
  };
};

export const nextSignupPage = () => {
  return {
    type: MOVE_SIGNUP_FORM_FORWARD
  };
};

export const previousSignupPage = () => {
  return {
    type: MOVE_SIGNUP_FORM_BACKWARD
  };
};

export const closeSignUpModal = () => {
  return {
    type: CLOSE_SIGNUP_MODAL
  };
};

//============================================================== Login Action Creators
export const loginAccount = creds => dispatch => {
  dispatch({ type: LOGIN_APP_START });
  axios
    .post("https://burning-heart.herokuapp.com/api/auth/login-account", creds)
    .then(res => {
      dispatch({ type: LOGIN_APP_SUCCESS, payload: res.data });
    })
    .catch(err => {
      if (err.response.data.message === restrictedError) {
        dispatch({ type: EXPIRED_CREDENTIALS, payload: err }); // Move later for other API calls... Not relevant here, but preserving code
      } else {
        dispatch({ type: LOGIN_APP_ERROR, payload: err });
      }
    });
};

export const loginCharity = creds => dispatch => {
  dispatch({ type: LOGIN_CHARITY_START });
  axios
    .post("https://burning-heart.herokuapp.com/api/auth/login-charity", creds)
    .then(res => {
      dispatch({ type: LOGIN_CHARITY_SUCCESS, payload: res.data });
    })
    .catch(err => {
      if (err.response.data.message === restrictedError) {
        dispatch({ type: EXPIRED_CREDENTIALS, payload: err }); // Move later for other API calls... Not relevant here, but preserving code
      } else {
        dispatch({ type: LOGIN_CHARITY_ERROR, payload: err });
      }
    });
};

export const loginWithToken = _ => dispatch => {
  dispatch({ type: LOGIN_TOKEN_START });
  const token = localStorage.getItem("bhToken");
  let decodedToken = "";

  if (token) {
    decodedToken = decode(token);
  }

  const reqOptions = {
    headers: { authorization: token }
  };

  axios
    .get(
      `https://burning-heart.herokuapp.com/api/restricted/accounts/${
        decodedToken.subject
      }`,
      reqOptions
    )
    .then(res => {
      dispatch({ type: LOGIN_TOKEN_SUCCESS, payload: res.data });
    })
    .catch(err => {
      dispatch({ type: LOGIN_TOKEN_ERROR, payload: err });
    });
};

export const loginWithTokenCharity = _ => dispatch => {
  dispatch({ type: LOGIN_TOKEN_CHARITY_START });
  const token = localStorage.getItem("bhToken");
  let decodedToken = "";

  if (token) {
    decodedToken = decode(token);
  }

  const reqOptions = {
    headers: { authorization: token }
  };

  axios
    .get(
      `https://burning-heart.herokuapp.com/api/restricted/charities/${
        decodedToken.subject
      }`,
      reqOptions
    )
    .then(res => {
      dispatch({ type: LOGIN_TOKEN_CHARITY_SUCCESS, payload: res.data });
    })
    .catch(err => {
      dispatch({ type: LOGIN_TOKEN_CHARITY_ERROR, payload: err });
    });
};

export const handleLoginForm = (name, value) => {
  return {
    type: UPDATE_LOGIN_FORM,
    payload: { name, value }
  };
};

export const closeLoginModal = () => {
  return {
    type: CLOSE_LOGIN_MODAL
  };
};

export const logoutApp = () => {
  return {
    type: LOGOUT_APP
  };
};

//============================================================== Donor Account Action Creators

export const fetchAccountDonations = _ => dispatch => {
  dispatch({ type: FETCH_USER_DONATIONS_START });
  const token = localStorage.getItem("bhToken");
  let decodedToken = "";
  if (token) {
    decodedToken = decode(token);
  }
  const reqOptions = {
    headers: { authorization: token }
  };
  axios
    .get(
      `https://burning-heart.herokuapp.com/api/restricted/donations/account/${
        decodedToken.subject
      }`,
      reqOptions
    )
    .then(res => {
      dispatch({ type: FETCH_USER_DONATIONS_SUCCESS, payload: res.data });
    })
    .catch(err => {
      dispatch({ type: FETCH_USER_DONATIONS_ERROR, payload: err });
    });
};

export const fetchCharityDonations = _ => dispatch => {
  dispatch({ type: FETCH_CHARITY_DONATIONS_START });
  const token = localStorage.getItem("bhToken");
  let decodedToken = "";
  if (token) {
    decodedToken = decode(token);
  }
  const reqOptions = {
    headers: { authorization: token }
  };
  axios
    .get(
      `https://burning-heart.herokuapp.com/api/restricted/donations/charity/${
        decodedToken.subject
      }`,
      reqOptions
    )
    .then(res => {
      dispatch({ type: FETCH_CHARITY_DONATIONS_SUCCESS, payload: res.data });
    })
    .catch(err => {
      dispatch({ type: FETCH_CHARITY_DONATIONS_ERROR, payload: err });
    });
};

export const fetchCharities = _ => dispatch => {
  dispatch({ type: FETCH_CHARITIES_START });
  const token = localStorage.getItem("bhToken");
  const reqOptions = {
    headers: { authorization: token }
  };
  axios
    .get(
      "https://burning-heart.herokuapp.com/api/restricted/charities",
      reqOptions
    )
    .then(res => {
      const sortedCharities = res.data.charities.sort(function(a, b) {
        if (a.charityName > b.charityName) return 1;
        if (a.charityName < b.charityName) return -1;
        return 0;
      });
      const sortedCharityRes = {
        ...res.data,
        charities: sortedCharities
      };
      dispatch({ type: FETCH_CHARITIES_SUCCESS, payload: sortedCharityRes });
    })
    .catch(err => {
      dispatch({ type: FETCH_CHARITIES_ERROR, payload: err });
    });
};

export const makePayment = (
  amount,
  cardToken,
  charityId,
  accountId
) => dispatch => {
  dispatch({ type: CARD_PAYMENT_START });
  const donationData = {
    donation: { amount },
    stripeData: { tokenId: cardToken.token.id },
    charityId: Number(charityId),
    accountId
  };
  const token = localStorage.getItem("bhToken");
  const reqOptions = { headers: { authorization: token } };
  return axios
    .post(
      "https://burning-heart.herokuapp.com/api/restricted/donations",
      donationData,
      reqOptions
    )
    .then(res => {
      dispatch({ type: CARD_PAYMENT_SUCCESS, payload: res.data });
    })
    .catch(err => {
      dispatch({
        type: CARD_PAYMENT_ERROR,
        payload: err
      });
    });
};

export const handleCardTokenError = err => {
  return {
    type: HANDLE_CARD_TOKEN_ERROR,
    payload: err
  };
};

export const closePaymentModal = () => {
  return {
    type: CLOSE_PAYMENT_MODAL
  };
};

export const handlePaymentForm = (name, value) => {
  return {
    type: HANDLE_PAYMENT_FORM,
    payload: { name, value }
  };
};

export const selectCharity = charityId => {
  return { type: SELECT_CHARITY, payload: charityId };
};

export const selectDonationsList = () => {
  return { type: SELECT_DONATIONS };
};

export const selectCharitiesList = () => {
  return { type: SELECT_CHARITIES };
};
