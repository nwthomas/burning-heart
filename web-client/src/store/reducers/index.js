import { reducer } from "./reducer"; // Must import each type of reducer you use
import { combineReducers } from "redux"; // Import that allows you to combine reducers

// Combines reducers - don't forget to reference state by saying state.reducer.item
export default combineReducers({
  reducer
});
