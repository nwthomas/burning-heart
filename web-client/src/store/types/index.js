// =========================================================== Variable Types
export const VARIABLE_NAMES_STARTED = "VARIABLE_NAMES_STARTED";
export const VARIABLE_NAMES_SUCCESS = "VARIABLE_NAMES_SUCCESS";
export const VARIABLE_NAMES_ERROR = "VARIABLE_NAMES_ERROR";

// =========================================================== Insert More as Needed
