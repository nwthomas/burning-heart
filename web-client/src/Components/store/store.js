import React, { useReducer, createContext } from "react";
import { initialState, reducer } from "./reducers";

export const Store = createContext({
  state: {},
  dispatch: () => {}
});

export const MyStore = props => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Store.Provider value={{ state, dispatch }}>
      {props.children}
    </Store.Provider>
  );
};
