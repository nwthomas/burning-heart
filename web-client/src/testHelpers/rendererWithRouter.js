import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default rendererWithRouter;
