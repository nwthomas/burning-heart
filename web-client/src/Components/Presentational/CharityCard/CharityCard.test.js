import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../../testHelpers";
import { Provider } from "react-redux";
import { store } from "../../../store";

import CharityCard from "./CharityCard";

const charity = { id: 1, charityName: "Charity" };
const CharityCardWithStore = () => {
  return (
    <Provider store={store}>
      <CharityCard charity={charity} />
    </Provider>
  );
};

describe("<CharityCard />", () => {
  it("should render the CharityCard without crashing", () => {
    const helpers = renderWithRouter(<CharityCardWithStore />);
  });

  it("should match the snapshot of CharityCard", () => {
    const tree = rendererWithRouter(<CharityCardWithStore />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
