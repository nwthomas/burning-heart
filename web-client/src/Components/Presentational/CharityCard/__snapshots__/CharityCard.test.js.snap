// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharityCard /> should match the snapshot of CharityCard 1`] = `
<div
  className="charity__card"
>
  <p
    className="charity__name"
  >
    Charity
  </p>
  <button
    className="charity__donate-button"
    onClick={[Function]}
    type="button"
  >
    Donate
  </button>
</div>
`;
